compiler/front/msgs.nim

```diff
@@ -516,6 +516,535 @@ proc handleReport*(
+  of adSemWrongIdent: rsemWrongIdent
+  of adSemPragmaOptionExpected: rsemPragmaOptionExpected
+  of adSemUnexpectedPushArgument: rsemUnexpectedPushArgument
+  of adSemMismatchedPopPush: rsemMismatchedPopPush
+  of adSemExcessiveCompilePragmaArgs: rsemExcessiveCompilePragmaArgs
+  of adSemEmptyAsm: rsemEmptyAsm
+  of adSemAsmEmitExpectsStringLiteral: rsemIllformedAst
+  of adSemLinePragmaExpectsTuple: rsemLinePragmaExpectsTuple
+  of adSemRaisesPragmaExpectsObject: rsemRaisesPragmaExpectsObject
+  of adSemLocksPragmaExpectsList: rsemLocksPragmaExpectsList
+  of adSemLocksPragmaBadLevelRange: rsemLocksPragmaBadLevel
+  of adSemLocksPragmaBadLevelString: rsemLocksPragmaBadLevel
+  of adSemBorrowPragmaNonDot: rsemBorrowPragmaNonDot
+  of adSemInvalidExtern: rsemInvalidExtern
+  of adSemBadDeprecatedArg: rsemBadDeprecatedArg
+  of adSemBadDeprecatedArgs: rsemBadDeprecatedArgs
+  of adSemMisplacedEffectsOf: rsemMisplacedEffectsOf
+  of adSemMissingPragmaArg: rsemMissingPragmaArg
+  of adSemCannotPushCast: rsemCannotPushCast
+  of adSemCastRequiresStatement: rsemCastRequiresStatement
+  of adSemPragmaRecursiveDependency: rsemPragmaRecursiveDependency
+  of adSemImportjsRequiresJs: rsemImportjsRequiresJs
+  of adSemBitsizeRequires1248: rsemBitsizeRequires1248
+  of adSemAlignRequiresPowerOfTwo: rsemAlignRequiresPowerOfTwo
+  of adSemNoReturnHasReturn: rsemNoReturnHasReturn
+  of adSemMisplacedDeprecation: rsemMisplacedDeprecation
+  of adSemCustomUserError: rsemCustomUserError
+  of adSemFatalError: rsemFatalError
+  of adSemNoUnionForJs: rsemNoUnionForJs
+  of adSemBitsizeRequiresPositive: rsemBitsizeRequiresPositive
+  of adSemExperimentalRequiresToplevel: rsemExperimentalRequiresToplevel
+  of adSemImplicitPragmaError: rsemImplicitPragmaError
+  of adSemPragmaDynlibRequiresExportc: rsemPragmaDynlibRequiresExportc
+  of adSemIncompatibleDefaultExpr: rsemIncompatibleDefaultExpr
+  of adSemWrongNumberOfArguments: rsemWrongNumberOfArguments
+  of adVmUnsupportedNonNil: rvmUnsupportedNonNil
+  of adVmDerefNilAccess: rvmNilAccess
+  of adVmDerefAccessOutOfBounds: rvmAccessOutOfBounds
+  of adVmDerefAccessTypeMismatch: rvmAccessTypeMismatch
+  of adCyclicTree: rsemCyclicTree
+  of adVmGenError:
+    assert vmGenDiag.isSome
+    astDiagVmGenToLegacyReportKind(vmGenDiag.unsafeGet)
+  of adVmError:
+    assert vmEvent.isSome
+    astDiagVmToLegacyReportKind(vmEvent.unsafeGet)
+  of adVmQuit: rvmQuit
+  of adSemRawTypeMismatch: rsemRawTypeMismatch
+  of adSemExpressionCannotBeCalled: rsemExpressionCannotBeCalled
+  of adSemCallTypeMismatch: rsemCallTypeMismatch
+
+
+func astDiagToLegacyReport*(diag: PAstDiag): Report {.inline.} =
+  let
+    kind =
+      case diag.kind
+      of adVmGenError:
+        astDiagToLegacyReportKind(diag.kind, some diag.vmGenErr.kind)
+      else:
+        astDiagToLegacyReportKind(diag.kind)
+    rep =
+      case diag.kind
+      of adWrappedError:
+        SemReport(
+            location: std_options.some diag.location,
```

Comment:
I believe the `some` identifier doesn't need to be qualified both here and for the other occurrences. Feel free to disregard if you did this for stylistic reasons.

Reply:
I left them in because they're not pretty, trying to encourage "it's ugly code, so fix more reports stuff so it goes away". :D